arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -538,3 +545,28 @@ public fun <A : AutoCloseable> autoCloseable(
 ): Resource<A> = resource {
   autoCloseable(closingDispatcher, autoCloseable)
 }
+
+private fun CompletableJob.completeWith(exitCase: ExitCase) {
+  when (exitCase) {
+    is ExitCase.Cancelled -> cancel(exitCase.exception)
+    is ExitCase.Failure -> completeExceptionally(exitCase.failure)
+    ExitCase.Completed -> complete()
+  }
+}
+
+private suspend fun CompletableJob.completeWithAndJoin(exitCase: ExitCase) {
+  completeWith(exitCase)
+  return join()
+}
```

Comment:
This is more of a style thing, but may you (literally) inline this function into its (single) usage below. I think that it would made for a simpler codebase.

Reply:
Sure - I split it to mirror `Job`'s `cancel` and `cancelWithJoin` but as it's not exposed now there's no benefit.